src/relentless/model/potential/dihedral.py

```diff
@@ -0,0 +1,482 @@
+        if s:
+            d = d.item()
+        return d
+
+
+class RyckaertBellemansDihedral(DihedralPotential):
+    r"""Cosine squared dihedral potential.
+
+    .. math::
+
+        u(\phi) = c_0 + c_1 (\cos (\phi - \pi)) + c_2 (\cos (\phi - \pi))^2 +
+        c_3 (\cos (\phi - \pi))^3 + c_4 (\cos (\phi - \pi))^4 +
+        c_5 (\cos (\phi - \pi))^5
+
+    where :math:`\phi` is the dihedral between four bonded particles. The parameters
+    for each type are:
+
+    +-------------+--------------------------------------------------+
+    | Parameter   | Description                                      |
+    +=============+==================================================+
+    | ``c_0``     | First fitting coefficient :math:`c_0`.           |
+    +-------------+--------------------------------------------------+
+    | ``c_1``     | Second fitting coefficient :math:`c_1`.          |
+    +-------------+--------------------------------------------------+
+    | ``c_2``     | Third fitting coefficient :math:`c_2`.           |
+    +-------------+--------------------------------------------------+
+    | ``c_3``     | Fourth fitting coefficient :math:`c_3`.          |
+    +-------------+--------------------------------------------------+
+    | ``c_4``     | Fifth fitting coefficient :math:`c_4`.           |
+    +-------------+--------------------------------------------------+
+    | ``c_5``     | Sixth fitting coefficient :math:`c_5`.           |
+    +-------------+--------------------------------------------------+
+
+    Parameters
+    ----------
+    types : tuple[str]
+        Types.
+    name : str
+        Unique name of the potential. Defaults to ``__u[id]``, where ``id`` is the
+        unique integer ID of the potential.
+
+    Attributes
+    ----------
+    coeff : :class:`DihedralParameters`
+        Parameters of the potential for each type.
+
+    Examples
+    --------
+    Harmonic Dihedral::
+
+        >>> u = relentless.potential.dihedral.RyckaertBellemansDihedral(("A",))
+        >>> u.coeff["A"].update({
+            'c0': 9.28,
+            'c1': 12.16,
+            'c2': -13.12,
+            'c3': -3.06,
+            'c4': 26.24,
+            'c5': -31.5
+            })
+
+    """
+
+    def __init__(self, types, name=None):
+        super().__init__(
+            keys=types, params=("c0", "c1", "c2", "c3", "c4", "c5"), name=name
+        )
+
+    def energy(self, type_, phi):
+        """Evaluate dihedral energy."""
+        params = self.coeff.evaluate(type_)
+        c0 = params["c0"]
+        c1 = params["c1"]
+        c2 = params["c2"]
+        c3 = params["c3"]
+        c4 = params["c4"]
+        c5 = params["c5"]
+
+        phi, u, s = self._zeros(phi)
+
+        psi = phi - numpy.pi
+
+        u = (
+            c0
+            + c1 * numpy.cos(psi)
+            + c2 * numpy.cos(psi) ** 2
+            + c3 * numpy.cos(psi) ** 3
+            + c4 * numpy.cos(psi) ** 4
+            + c5 * numpy.cos(psi) ** 5
```

Comment:
Performance isn't a HUGE concern for us, but I would at least store `cos_phi` in a variable and use it to evaluate these powers, so the cosine isn't calculated 6 times.